clang/test/CodeGenSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,138 @@
+
+        DEF_IN_MACRO();
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[MACRO_SIZE]], [[MACRO_SIZE]]* @[[MACRO_X]]
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[MACRO_SIZE]], [[MACRO_SIZE]]* @[[MACRO_Y]]
+
+        MACRO_CALLS_MACRO();
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[MACRO_MACRO_SIZE]], [[MACRO_MACRO_SIZE]]* @[[MACRO_MACRO_X]]
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[MACRO_MACRO_SIZE]], [[MACRO_MACRO_SIZE]]* @[[MACRO_MACRO_Y]]
+
+        template_param<int>();
+        // CHECK: define linkonce_odr spir_func void @_Z14template_paramIiEvv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[INT_SIZE]], [[INT_SIZE]]* @[[INT]]
+
+        template_param<decltype(x)>();
+        // CHECK: define internal spir_func void @"_Z14template_paramIZZ4mainENK3
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[LAMBDA_X_SIZE]], [[LAMBDA_X_SIZE]]* @[[LAMBDA_X]]
+
+        lambda_in_dependent_function<int>();
+        // CHECK: define linkonce_odr spir_func void @_Z28lambda_in_dependent_functionIiEvv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_INT_SIZE]], [[DEP_INT_SIZE]]* @[[LAMBDA_IN_DEP_INT]]
+
+        lambda_in_dependent_function<decltype(x)>();
+        // CHECK: define internal spir_func void @_Z28lambda_in_dependent_functionIZZ4mainENK3$_0clEvEUlvE_Evv
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_LAMBDA_SIZE]], [[DEP_LAMBDA_SIZE]]* @[[LAMBDA_IN_DEP_X]]
+
+        lambda_no_dep<int, double>(3, 5.5);
+        // CHECK: define linkonce_odr spir_func void @_Z13lambda_no_depIidEvT_T0_(i32 %a, double %b)
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([38 x i8], [38 x i8]* @usn_str.13, i32 0, i32 0)) #1
+        //  call spir_func void @printf(i8* getelementptr inbounds ([[NO_DEP_LAMBDA_SIZE]], [[NO_DEP_LAMBDA_SIZE]]* [[LAMBDA_NO_DEP]]
+
+        int a = 5;
+        double b = 10.7;
+        auto y = [](int a) { return a; };
+        auto z = [](double b) { return b; };
+        lambda_two_dep<decltype(y), decltype(z)>();
+        // CHECK: define internal spir_func void @"_Z14lambda_two_depIZZ4mainENK3$_0clEvEUliE_ZZ4mainENKS0_clEvEUldE_Evv"
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_LAMBDA1_SIZE]], [[DEP_LAMBDA1_SIZE]]* [[LAMBDA_TWO_DEP]]
+
+        lambda_two_dep<decltype(z), decltype(y)>();
+        // CHECK: define internal spir_func void @"_Z14lambda_two_depIZZ4mainENK3$_0clEvEUldE_ZZ4mainENKS0_clEvEUliE_Evv"()
+        // CHECK: call spir_func void @printf(i8* getelementptr inbounds ([[DEP_LAMBDA2_SIZE]], [[DEP_LAMBDA2_SIZE]]* [[LAMBDA_TWO_DEP]]
+      });
+#ifdef DERROR
+  // CHECK: expected-error@+1{{unique-stable-name mangling not implemented yet}}
+  kernel_single_task<class kernel2>(func<Derp>());
```

Comment:
Changing this case to check IR, now that Erich implemented mangling for trailing return types